lightning/src/ln/channel.rs

```diff
@@ -1263,7 +1278,9 @@ impl<Signer: Sign> Channel<Signer> {
 			}
 		}
 		if pending_idx == core::usize::MAX {
-			return Err(ChannelError::Ignore("Unable to find a pending HTLC which matched the given HTLC ID".to_owned()));
+			#[cfg(any(test, feature = "fuzztarget"))]
+			debug_assert!(self.historical_inbound_htlc_fulfills.contains(&htlc_id_arg));
```

Comment:
Maybe update comment above about "ChannelManager may generate duplicate claims/fails due to HTLC update events from..." to also mention duplicative claims due to peer connect/disconnect ?